### PR TITLE
tests: mem_protect: add error test case of userspace

### PR DESCRIPTION
**Pull request by enjiamai**
Add some error test cases for userspace of mem protect module, 
in order to improve better code coverage in our module testing.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>

### CODE REVIEW

File: tests/kernel/mem_protect/mem_map/src/main.c [hunk 1]
```diff
@@ -165,13 +165,28 @@ void test_z_phys_map_side_effect(void)
 	ztest_test_fail();
 }
 
+/**
+ * Show that mapping works failed as expected
```
Comment (LeiW000) [inline tests/kernel/mem_protect/mem_map/src/main.c:169]: Please add "@brief" at begining of the line.

File: tests/kernel/mem_protect/userspace/src/main.c [hunk 2]
```diff
@@ -528,13 +528,13 @@ static void umode_enter_func(void)
...
 
 /**
-* @brief Test to check supervisor thread enter one-way to usermode
-*
-* @details A thread running in supervisor mode must have one-way operation
-* ability to drop privileges to user mode.
-*
-* @ingroup kernel_memprotect_tests
-*/
+ * @brief Test to check supervisor thread enter one-way to usermode
```
Comment (LeiW000) [inline tests/kernel/mem_protect/userspace/src/main.c:531]: It better to change it like below.
Check if supervisoer thread can enter usermode with one-way operation.

**Review by LeiW000 — CHANGES_REQUESTED** [review]

File: tests/kernel/mem_protect/userspace/src/main.c (hunk 2 above)
Comment (enjiamai) [reply tests/kernel/mem_protect/userspace/src/main.c:531]: @LeiW000  Sorry for reply late. Got it fix, thanks!

File: tests/kernel/mem_protect/mem_map/src/main.c (hunk 1 above)
Comment (enjiamai) [reply tests/kernel/mem_protect/mem_map/src/main.c:169]: @LeiW000 Got it fix, this test case conflict due to another test case merged. I will put them in another commit.

File: tests/kernel/mem_protect/userspace/src/main.c [hunk 3]
```diff
@@ -912,6 +913,381 @@ void test_tls_pointer(void)
...
+ * generate an expected out of memory error.
+ *
+ * @ingroup kernel_memprotect_tests
+ */
+void test_dynamic_obj_create_out_of_memory(void)
+{
+	struct z_object *ko;
+
+	ko = z_dynamic_object_create(INT_MAX);
+	zassert_is_null(ko, "success is not reasonable");
```
Comment (dcpleung) [inline tests/kernel/mem_protect/userspace/src/main.c:1236]: This message is confusing if you only look at serial output when this fails. Please state exactly what is wrong.

**Review by dcpleung — CHANGES_REQUESTED** [review]

File: tests/kernel/mem_protect/userspace/src/main.c [hunk 4]
```diff
@@ -912,6 +913,381 @@ void test_tls_pointer(void)
...
+	struct k_stack *s;
+	struct k_pipe *p;
+	struct k_queue *q;
+	struct k_mem_slab *mslab;
+	struct k_poll_signal *polls;
+	struct k_timer *timer;
+	void *ko;
+	struct k_thread *curr_thread = k_current_get();
+
+#if !defined(CONFIG_X86_64)
```
[the remark this answers is not in the document]
Comment (enjiamai) [reply tests/kernel/mem_protect/userspace/src/main.c:1149]: Hi, it seems like that k_obj_alloc(K_OBJ_THREAD) still return NULL, give me more time to check it, thanks!

File: tests/kernel/mem_protect/userspace/src/main.c (hunk 3 above)
Comment (enjiamai) [reply tests/kernel/mem_protect/userspace/src/main.c:1236]: Got it, indeed it is confusing. Thanks for correction this^^

**Review by LeiW000 — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**